stixcore/config/reader.py

```diff
@@ -51,6 +56,89 @@ def int_def(value, default=0):
         return default
 
 
+def get_sci_channels(date):
+    r"""
+    Get the science energy channel info for given date
+
+    Parameters
+    ----------
+    date : `datetime.datetime`
+
+    Returns
+    -------
+    `astropy.table.QTable`
+        Science Energy Channels
+    """
+    root = Path(__file__).parent.parent
+    sci_chan_index_file = Path(root, *['config', 'data', 'common',
+                                       'detector', 'science_echan_index.csv'])
+
+    sci_chan_index = read_energy_channel_index(sci_chan_index_file)
+    sci_info = sci_chan_index.at(date)
+    if len(sci_info) == 0:
+        raise ValueError(f'No Science Energy Channel file found for date {date}')
+    elif len(sci_info) > 1:
+        raise ValueError(f'Multiple Science Energy Channel file for date {date}')
+    start_date, end_date, sci_echan_file = list(sci_info)[0]
+    sci_echan_table = read_sci_energy_channels(sci_echan_file)
+
+    return sci_echan_table
+
+
```

Comment:
Well with the current version:
```
 %timeit get_sci_channels(datetime.datetime(2022, 12, 1))
19.6 ms ± 1.75 ms per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

with a caching version 
```
In [3]: %timeit get_sci_channels(datetime.datetime(2022, 12, 1))
setting index
7.15 ms ± 677 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
``` 

```
def get_sci_channels(date):
    r"""
    Get the science energy channel info for given date

    Parameters
    ----------
    date : `datetime.datetime`

    Returns
    -------
    `astropy.table.QTable`
        Science Energy Channels
    """
    global SCI_INDEX
    if SCI_INDEX is None:
        print('setting index')
        root = Path(__file__).parent.parent
        sci_chan_index_file = Path(root, *['config', 'data', 'common',
                                       'detector', 'science_echan_index.csv'])

        sci_chan_index = read_energy_channel_index(sci_chan_index_file)
        SCI_INDEX = sci_chan_index

    sci_info = SCI_INDEX.at(date)
    if len(sci_info) == 0:
        raise ValueError(f'No Science Energy Channel file found for date {date}')
    elif len(sci_info) > 1:
        raise ValueError(f'Multiple Science Energy Channel file for date {date}')
    start_date, end_date, sci_echan_file = list(sci_info)[0]
    sci_echan_table = read_sci_energy_channels(sci_echan_file)

    return sci_echan_table
```

Reply:
sorry for being picky now you cach just the index what is already good.
But i think you should also cache the read ELUT if read once.
you use the path of the file for the entry in a hash map. if the key is not present you read the LUT and store the structore in that dict otherwise you return it directly.
it is very likely that during on one processing batch just one elut is read again and again ....

Reply:
No sure this is the point of the review!

With current implementation caching both the index and sci channel dow to microseconds
```
%timeit get_sci_channels(datetime.datetime(2022, 12, 1))
setting index
3.25 µs ± 725 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```

Reply:
very nice